incubator/xray/templates/_helpers.tpl

```diff
@@ -0,0 +1,158 @@
+{{- .Values.analysis.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- $name := default .Chart.Name .Values.analysis.name -}}
+{{- if contains $name .Release.Name -}}
+{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "xray-indexer.fullname" -}}
+{{- if .Values.indexer.fullnameOverride -}}
+{{- .Values.indexer.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- $name := default .Chart.Name .Values.indexer.name -}}
+{{- if contains $name .Release.Name -}}
+{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "xray-persist.fullname" -}}
+{{- if .Values.persist.fullnameOverride -}}
+{{- .Values.persist.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- $name := default .Chart.Name .Values.persist.name -}}
+{{- if contains $name .Release.Name -}}
+{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "xray-server.fullname" -}}
+{{- if .Values.server.fullnameOverride -}}
+{{- .Values.server.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- $name := default .Chart.Name .Values.server.name -}}
+{{- if contains $name .Release.Name -}}
+{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Set the final MongoDB connection URL
+*/}}
+{{- define "xray.mongodb.url" -}}
+{{- if .Values.global.mongoUrl -}}
+{{- .Values.global.mongoUrl -}}
+{{- else -}}
+{{- printf "mongodb://xray:password@%s-mongodb:27017/?authSource=xray&authMechanism=SCRAM-SHA-1" .Release.Name -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Set the final PostgreSQL connection URL
+*/}}
+{{- define "xray.postgresql.url" -}}
+{{- if .Values.global.postgresqlUrl -}}
+{{- .Values.global.postgresqlUrl -}}
+{{- else -}}
+{{- printf "postgres://xray:xray@%s-postgresql:5432/xraydb?sslmode=disable" .Release.Name -}}
```

Comment:
Don't hard-code credentials.

Reply:
@unguiculus same for postgres and rabbitmq-ha